### ISSUE TITLE
Prometheus fails to scrape uyuni clients - Uyuni 2022.02

### ISSUE THREAD

**Issue by gvaynapel** [uyuni-project/uyuni#4794]
# Additional Information

_The following information is very important in order to help us help you. Omission of the following details can cause delays or can receive no attention at all._

## Version of Uyuni Server and Proxy (if used)
`zypper info Uyuni-Server-release` _and_ `zypper info Uyuni-Proxy-release` _(if used)_

Information for package Uyuni-Server-release:
---------------------------------------------
Repository     : uyuni-server-stable
Name           : Uyuni-Server-release
Version        : 2022.01-165.2.uyuni1
Arch           : x86_64
Vendor         : obs://build.opensuse.org/systemsmanagement:Uyuni
Support Level  : Level 3
Installed Size : 1.4 KiB
Installed      : Yes (automatically)
Status         : out-of-date (version 2021.06-132.3.uyuni1 installed)
Source package : Uyuni-Server-release-2022.01-165.2.uyuni1.src
Summary        : Uyuni Server
Description    :
    Uyuni lets you efficiently manage physical, virtual,
    and cloud-based Linux systems. It provides automated and cost-effective
    configuration and software management, asset management, and system
    provisioning.

## Details about issue
_Include as many details as you can. The procedure you are following,  the commands you are running, logs (specially from_ `/var/log/rhn/`_)..._
_In general: all that can help someone to reproduce the issue or understand it_

Prometheus version: prom/prometheus:latest (I believe it is 2.33.1)

Prometheus exporter formula deployed to a managed system. Prometheus server is unable to scrape the logs and returns this error:

```
prometheus_1  | ts=2022-02-08T05:37:04.390Z caller=refresh.go:98 level=error component="discovery manager scrape" discovery=uyuni msg="Unable to refresh target groups" err="unable to get endpoints information: Fault(-1): redstone.xmlrpc.XmlRpcFault: unhandled internal exception: null"
prometheus_1  | ts=2022-02-08T05:38:04.363Z caller=refresh.go:98 level=error component="discovery manager scrape" discovery=uyuni msg="Unable to refresh target groups" err="unable to get endpoints information: Fault(-1): redstone.xmlrpc.XmlRpcFault: unhandled internal exception: null"
prometheus_1  | ts=2022-02-08T05:39:04.401Z caller=refresh.go:98 level=error component="discovery manager scrape" discovery=uyuni msg="Unable to refresh target groups" err="unable to get endpoints information: Fault(-1): redstone.xmlrpc.XmlRpcFault: unhandled internal exception: null"
prometheus_1  | ts=2022-02-08T05:40:04.394Z caller=refresh.go:98 level=error component="discovery manager scrape" discovery=uyuni msg="Unable to refresh target groups" err="unable to get endpoints information: Fault(-1): redstone.xmlrpc.XmlRpcFault: unhandled internal exception: null"
prometheus_1  | ts=2022-02-08T05:41:04.441Z caller=refresh.go:98 level=error component="discovery manager scrape" discovery=uyuni msg="Unable to refresh target groups" err="unable to get endpoints information: Fault(-1): redstone.xmlrpc.XmlRpcFault: unhandled internal exception: null"
prometheus_1  | ts=2022-02-08T05:42:04.395Z caller=refresh.go:98 level=error component="discovery manager scrape" discovery=uyuni msg="Unable to refresh target groups" err="unable to get endpoints information: Fault(-1): redstone.xmlrpc.XmlRpcFault: unhandled internal exception: null"
```

Prometheus config:

```
scrape_configs:

  # Make Prometheus scrape itself for metrics.
  - job_name: 'prometheus'
    static_configs:
      - targets: ['localhost:9090']
  - job_name: 'mgr-server'
    static_configs:
      - targets:
        - 'uyuni.internal.domain.com:9100'  # Node exporter
        - 'uyuni.internal.domain.com:9187'  # PostgreSQL exporter
        - 'uyuni.internal.domain.com:5556'  # JMX exporter (Tomcat)
        - 'uyuni.internal.domain.com:5557'  # JMX exporter (Taskomatic)
        - 'uyuni.internal.domain.com:9800'  # Taskomatic
      - targets:
        - 'uyuni.internal.domain.com:80'    # Message queue
        labels:
          __metrics_path__: /rhn/metrics

 - job_name: 'mgr-client'

    uyuni_sd_configs:
      - server: http://uyuni.internal.domain.com
        username: username
        password: notmyrealpassword

    relabel_configs:
      - source_labels: [__meta_uyuni_exporter]
        target_label: exporter
      - source_labels: [__meta_uyuni_groups]
        target_label: groups
      - source_labels: [__meta_uyuni_minion_hostname]
        target_label: hostname
      - source_labels: [__meta_uyuni_primary_fqdn]
        regex: (.+)
        target_label: hostname
      - source_labels: [hostname, __address__]
        regex: (.*);.*:(.*)
        replacement: ${1}:${2}
        target_label: __address__
      - source_labels: [__meta_uyuni_metrics_path]
        regex: (.+)
        target_label: __metrics_path__
      - source_labels: [__meta_uyuni_proxy_module]
        target_label: __param_module  
```                                                    

Formula Prometheus Details:

```
----------
          ID: exporter_exporter
    Function: service.dead
        Name: prometheus-exporter-exporter
      Result: true
     Comment: The named service prometheus-exporter-exporter is not available
     Started: 04:17:12.322535
    Duration: 26.736
         SLS: prometheus-exporters
     Changed: {}
----------
          ID: node_exporter
    Function: pkg.installed
        Name: prometheus-node-exporter
      Result: true
     Comment: 1 targeted package was installed/updated.
     Started: 04:17:12.350030
    Duration: 51308.478
         SLS: prometheus-exporters
     Changed: moreutils:
                  old: ''
                  new: 0.62-1
              guile-2.0-libs:
                  old: ''
                  new: 2.0.13+1-5ubuntu0.1
              mysql-common:
                  old: ''
                  new: 5.8+1.0.4
              prometheus-node-exporter:
                  old: ''
                  new: 0.17.0+ds-3
              libipc-run-perl:
                  old: ''
                  new: 0.96-1
              libntlm0:
                  old: ''
                  new: 1.4-8ubuntu0.1
              smartmontools:
                  old: ''
                  new: 6.5+svn4324-1ubuntu0.1
              libgsasl7:
                  old: ''
                  new: 1.8.0-8ubuntu3
              ssl-cert:
                  old: ''
                  new: 1.0.39
              libmysqlclient20:
                  old: ''
                  new: 5.7.37-0ubuntu0.18.04.1
              libgc1c2:
                  old: ''
                  new: 1:7.4.2-8ubuntu1
              mailutils-common:
                  old: ''
                  new: 1:3.4-1
              mailutils:
                  old: ''
                  new: 1:3.4-1
              libio-pty-perl:
                  old: ''
                  new: 1:1.08-1.1build4
              libmailutils5:
                  old: ''
                  new: 1:3.4-1
              postfix:
                  old: ''
                  new: 3.3.0-1ubuntu0.4
              libkyotocabinet16v5:
                  old: ''
                  new: 1.2.76-4.2
              
----------
          ID: node_exporter
    Function: file.managed
        Name: /etc/default/prometheus-node-exporter
      Result: true
     Comment: File /etc/default/prometheus-node-exporter updated
     Started: 04:18:03.662460
    Duration: 76.184
         SLS: prometheus-exporters
     Changed: diff: "--- 
              +++ 
              @@ -2,7 +2,7 @@
               # Due to shell scaping, to pass backslashes for regexes, you need to double
               # them (\\\\d for \\d). If running under\
                  \ systemd, you need to double them again
               # (\\\\\\\\d to mean \\d), and escape newlines too.
              -ARGS=\"\"
              +ARGS=--collector.systemd --web.listen-address=:9100
              \
                  \ 
               # Prometheus-node-exporter supports the following options:
               #
              "
              
----------
          ID: node_exporter
    Function: service.running
        Name: prometheus-node-exporter
      Result: true
     Comment: Service restarted
     Started: 04:18:05.535746
    Duration: 29.612
         SLS: prometheus-exporters
     Changed: prometheus-node-exporter: true
              
----------
          ID: apache_exporter
    Function: service.dead
        Name: prometheus-apache-exporter
      Result: true
     Comment: The named service prometheus-apache-exporter is not available
     Started: 04:18:05.565771
    Duration: 14.869
         SLS: prometheus-exporters
     Changed: {}
----------
          ID: postgres_exporter
    Function: service.dead
        Name: prometheus-postgres-exporter
      Result: true
     Comment: The named service prometheus-postgres-exporter is not available
     Started: 04:18:05.580975
    Duration: 11.774
         SLS: prometheus-exporters
     Changed: {}




```

**Comment by SSchott** [comment]
I thought I was going crazy with this, and that maybe one of my registered systems went south, but I have the exact same issue. It was actually working until I added some systems, and realized that the formulas of the Grafana/Prometheus Host were unticked within Uyuni, probably because I recently updated all my machines (even though they were working with older nodes). As I wanted to add new nodes and Uyuni to allow the autodiscovery, I reticked the boxes, and the autodiscovery now doesn't work for any node. Also, I noticed that the "Monitoring" Type is not added consistently to all nodes that have the Prometheus exporter enabled.

After adding the Monitoring type by hand, and as mentioned by @gvaynapel , in `/var/log/rhn/rhn_web_api.log` get:
```
CALL: formula.getCombinedFormulaDataByServerIds(prometheus, prometheus-exporters, [1000010010, 1000010007, 1000010044, 1000010004, 1000010003, 1000010001, 1000010038, 1000010005, 1000010043, 1000010002, 1000010006, 1000010042]) CALLER: (prometheus) TIME: 0.001 seconds
redstone.xmlrpc.XmlRpcFault: unhandled internal exception: null
```

Before it broke, instead of the internal exception I got a `CALL: system.getNetworkForSystems`, maybe that helps to narrow down the issue
